test/es6/proxyconstruction.js

```diff
@@ -0,0 +1,150 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+function testConstructProxy (object, shouldFail)
+{
+    let proxy1 = new Proxy(object, {construct: () => ({})});
+    let proxy2 = new Proxy(object, {});
+    
+    if (shouldFail)
+    {
+        assert.throws(() => {new proxy1 ()}, TypeError, "Function is not a constructor");
+        assert.throws(() => {new proxy2 ()}, TypeError, "Function is not a constructor");
+    }
+    else
+    {
+        assert.doesNotThrow(() => {new proxy1 ()});
+        assert.doesNotThrow(() => {new proxy2 ()});
+    }
+}
+
+
+var tests = [
+    {
+        name: "Proxy of anonymous Arrow function should not be constructable",
+        body() {
+            testConstructProxy(()=>{}, true);
+        }
+    },
+    {
+        name: "Proxy of named Arrow function should not be constructable",
+        body() {
+            let test2 = () => {}; 
+            testConstructProxy(test2, true);
+        }
+    },
+    {
+        name: "Proxy of anonymous Async function should not be constructable",
+        body() {
+            testConstructProxy(async function () {}, true);
+        }
+    },
+    {
+        name: "Proxy of named Async function should not be constructable",
+        body() {
+            testConstructProxy(async function test4() {}, true);
+        }
+    },
+    {
+        name: "Proxy of normal anonymous function should be constructable",
+        body() {
+            testConstructProxy(function () {}, false);
+        }
+    },
+    {
+        name: "Proxy of normal function should be constructable",
+        body() {
+            testConstructProxy(function test6 () {}, false);
+        }
+    },
+    {
+        name: "Proxy of static class member function should be constructable",
```

Comment:
nitpick: title says this should be constructable but we mean it should "not be" constructable, right?